src/storage.rs

```diff
@@ -29,19 +33,25 @@ lazy_static! {
             .unwrap_or(None)
             .unwrap_or(0)
     });
+    static ref CACHED_UNIT_AUTHORS: FifoCache<String, Vec<String>> =
+        FifoCache::with_capacity(config::MAX_ITEMS_IN_CACHE);
+    static ref CACHED_UNIT_WITNESSES: FifoCache<String, Vec<String>> =
+        FifoCache::with_capacity(config::MAX_ITEMS_IN_CACHE);
+    static ref _CACHED_ASSET_INFOS: FifoCache<String, Option<String>> =
+        FifoCache::with_capacity(config::MAX_ITEMS_IN_CACHE);
 }
 
 pub fn is_known_unit(unit: &String) -> bool {
-    CACHED_UNIT.read().unwrap().contains_key(unit) || KNOWN_UNIT.read().unwrap().contains(unit)
+    CACHED_UNIT.get(unit).is_some() || KNOWN_UNIT.get(unit).is_some()
 }
 
 pub fn set_unit_is_known(unit: &String) {
-    KNOWN_UNIT.write().unwrap().insert(unit.to_owned());
+    KNOWN_UNIT.insert(unit.to_owned(), true);
 }
 
-pub fn forget_unit(unit: &String) {
-    KNOWN_UNIT.write().unwrap().remove(unit);
-    CACHED_UNIT.write().unwrap().remove(unit);
+pub fn forget_unit(_unit: &String) {
+    /*  KNOWN_UNIT.write().unwrap().remove(unit);
+    CACHED_UNIT.write().unwrap().remove(unit); */
 
     unimplemented!()
```

Comment:
here we may not have the correct design.....
we still need to remove cached items explicitly. let's disscuss it later